src/vss_tools/vspec/vssexporters/vss2samm/vss2samm.py

```diff
@@ -0,0 +1,298 @@
+# Copyright (c) 2024 Contributors to COVESA
+#
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License 2.0 which is available at
+# https://www.mozilla.org/en-US/MPL/2.0/
+#
+# SPDX-License-Identifier: MPL-2.0
+
+#
+# Convert all vspec input files to a ESMF - Aspect Model Editor (SAMM) - ttl formatted file(s).
+#
+
+
+import importlib
+import logging
+from pathlib import Path
+
+import rich_click as click
+import vss_tools.vspec.cli_options as clo
+from vss_tools import log
+from vss_tools.vspec.main import get_trees
+from vss_tools.vspec.tree import VSSNode
+
+from .config import config as cfg
+
+VSSConcepts = None
+vss_helper = None
+ttl_helper = None
+
+
+def __setup_environment(output_namespace, vspec_version, split_depth: int) -> None:
```

Comment:
that setup function looks quite suspicious. Why do we need dynamic imports here?

Reply:
The goal here was to get user input, which then can be used to initialize the script config, and later to load the: samm_concepts, vss_helper and ttl_helper which are working with some of the config fields, which depend on the user input at the time of call of the script.

Again, this was just my understanding of handling this case, but if there is a better way to do it, I will be happy to do some refactoring :)

Reply:
Just use function/class arguments does not work? If not, you could have a global state such as `config.py` that contains objects that get filled at the script start and other modules can then import and use

Reply:
There is a call to **init** the config with user input data:

`
def __setup_environment(output_namespace, vspec_version, split_depth: int) -> None:
    # Initialize config before to load other helpers / libraries, based on defined by user input, config data
    cfg.init(output_namespace, vspec_version, split_depth)
`

without this, I was getting errors for undefined variables in the dynamically loaded **_helper**(s)